src/components/AprInfo/AprInfo.tsx

```diff
@@ -1,36 +1,58 @@
 import { Trans, t } from "@lingui/macro";
-import { formatDistanceToNowStrict } from "date-fns";
+import { addDays, formatDistanceToNowStrict, isPast } from "date-fns";
 import { useCallback, useMemo } from "react";
 
 import { getIncentivesV2Url } from "config/links";
 import { useLiquidityProvidersIncentives } from "domain/synthetics/common/useIncentiveStats";
 import { useLpAirdroppedTokenTitle } from "domain/synthetics/tokens/useAirdroppedTokenTitle";
 import { useChainId } from "lib/chains";
 import { formatAmount } from "lib/numbers";
+import { ENOUGH_DAYS_SINCE_LISTING_FOR_APY, getMarketListingDate } from "config/markets";
 
 import ExternalLink from "components/ExternalLink/ExternalLink";
 import StatsTooltipRow from "components/StatsTooltip/StatsTooltipRow";
 import Tooltip from "components/Tooltip/Tooltip";
 
 import sparkleIcon from "img/sparkle.svg";
 
+/**
+ * We let the APY to stabilize for a few days before showing it to the user
+ */
+function getIsBaseApyReadyToBeShown(listingDate: Date): boolean {
+  const enoughDateForApy = addDays(listingDate, ENOUGH_DAYS_SINCE_LISTING_FOR_APY);
+
+  return isPast(enoughDateForApy);
+}
+
+function getApyReadyToBeShownDate(listingDate: Date): Date {
+  return addDays(listingDate, ENOUGH_DAYS_SINCE_LISTING_FOR_APY);
+}
+
 export function AprInfo({
   apy,
   incentiveApr,
   showTooltip = true,
-  futureDateForApyReadiness,
+  tokenAddress,
 }: {
   apy: bigint | undefined;
   incentiveApr: bigint | undefined;
   showTooltip?: boolean;
-  /**
-   * @default undefined meaning that APY is ready to be shown
-   */
-  futureDateForApyReadiness?: Date;
+  tokenAddress: string;
 }) {
   const { chainId } = useChainId();
+
+  const listingDate = getMarketListingDate(chainId, tokenAddress);
+  const isBaseAprReadyToBeShown = getIsBaseApyReadyToBeShown(listingDate);
```

Comment:
lets move this line inside useMemo below? you anyway store `Date | undefined` in `apyReadyToBeShownDate` var

Reply:
fixed